src/storybook-components/SearchPage.vue

```diff
@@ -118,13 +121,60 @@ export default {
     // Methods for search results
     getSearchResults: function (value) {
       // Use your own custom code to handle it.
-      alert('Search for: "' + value + '"')
+      const split = value.split(' x')
+      if (split.length !== 2) {
+        this.searchFor = value
+      } else {
+        this.searchFor = split[0]
+        this.totalResultCount = parseInt(split[1], 10)
+      }
+      this.offset = 0
     },
 
     // Methods for search results
     pagerChange: function (newStep) {
       // Use your own custom code to handle it.
-      alert('Going to step: ' + newStep)
+      this.offset = (newStep - 1) * this.mock.searchResults.responseSize
+    }
+  },
+  computed: {
+    pagerOptions () {
+      const count = this.totalResultCount
+      const show = this.mock.searchResults.responseSize
+      return {
+        totalSteps: (count > show) ? (count / show) : 0,
+        initialStep: 1,
+        stepsAround: 2
+      }
+    },
+    subsetResults () {
+      let returnedResults = []
+      let step = 0
+      const totalSteps = this.mock.searchResults.responseSize
+      const start = this.offset
+      while (step < totalSteps) {
+        if (this.totalResults[start + step]) {
+          returnedResults.push(this.totalResults[start + step])
+          step++
+        } else {
+          break
+        }
+      }
+      return returnedResults
+    },
+    totalResults () {
+      if (this.totalResultCount === 0) {
+        return []
+      } else {
+        let returnedResults = []
+        const searchResultJSON = JSON.stringify(this.mock.searchResult)
+        for (let i = 0; i < this.totalResultCount; i++) {
+          let newResult = JSON.parse(searchResultJSON)
+          newResult.title = `${this.searchFor} result ${i + 1}`
+          returnedResults.push(newResult)
+        }
+        return returnedResults
+      }
```

Comment:
Hi @alan-cole I am confused for this part, are they not need to be inside our component? Does all the place we use search results need those logic?

Reply:
In order to ensure the issue has been fixed in ripple, we need a way to test that the search results and pagination functionality work. This means we need a way to simulate a search and changing between pages.

Before there was no such functionality in ripple - it would show static results that couldn't be altered and clicking pagination would show an alert, but not change page.

This logic is here for simulation but is not useful in a real world scenario. Nuxt uses different implementation to display real results.